src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -235,26 +238,71 @@ private static IEnumerable<string> GetAllowedChildren(Type type, ErrorSink error
             IEnumerable<string> allowedChildren,
             TagHelperDesignTimeDescriptor designTimeDescriptor)
         {
-            var requiredAttributes = GetCommaSeparatedValues(targetElementAttribute.Attributes);
+            var requiredAttributeDescriptors = GetRequiredAttributeDescriptors(targetElementAttribute);
 
             return BuildTagHelperDescriptor(
                 targetElementAttribute.Tag,
                 typeName,
                 assemblyName,
                 attributeDescriptors,
-                requiredAttributes,
+                requiredAttributeDescriptors,
                 allowedChildren,
                 targetElementAttribute.ParentTag,
                 targetElementAttribute.TagStructure,
                 designTimeDescriptor);
         }
 
+        private static IEnumerable<TagHelperRequiredAttributeDescriptor> GetRequiredAttributeDescriptors(
+            HtmlTargetElementAttribute targetElementAttribute)
+        {
+            var separatedAttributes = GetCommaSeparatedValues(targetElementAttribute.Attributes);
```

Comment:
Commas are allowed in quoted attribute values. So `GetCommaSeparatedValues()` no longer does the Right Thing:tm:. Likely need to parse the whole string in one pass. That would also help detect errors like quotes within a quoted string.